src/cmd/genTxnBodiesFromS3.cpp

```diff
@@ -0,0 +1,175 @@
+      return SUCCESS;
+    }
+    po::notify(vm);
+
+    string remoteS3Path = "s3://" + bucketName + "/" + backupFolderName;
+    string localBackupPath =
+        bfs::current_path().string() + "/" + backupFolderName;
+    // Download from S3
+    if (!SysCommand::ExecuteCmd(
+            SysCommand::WITHOUT_OUTPUT,
+            GetAwsS3CpString(remoteS3Path, localBackupPath))) {
+      LOG_GENERAL(WARNING, "Failed to download backup folder from S3 : s3://"
+                               << bucketName << "/" << backupFolderName);
+      return ERROR_DOWNLOADING_BACKUP;
+    } else {
+      LOG_GENERAL(DEBUG, "Backup folder downloaded successfully : ");
+    }
+
+    // create json output folder
+    if (saveToJsonFormat) {
+      if (!(boost::filesystem::exists(jsonOutputPath))) {
+        if (!boost::filesystem::create_directories(jsonOutputPath)) {
+          cerr << "Failed to created JSON output folder ! will skip creating "
+                  "txns in json format"
+               << endl;
+          saveToJsonFormat = false;
+        }
+      }
+    }
+
+    // Loop through all files in backupFolderName and store to leveldb
+    vector<string> listOfTxnFiles = getAllFilesInDir(localBackupPath);
+    for (const auto& txns_filename : listOfTxnFiles) {
+      ifstream infile;
+      LOG_GENERAL(INFO, "Parsing " << txns_filename << endl);
+      try {
+        // Now read back file to see if the TransactionWithReceipt is good
+        infile.open(txns_filename, ios::in | ios::binary);
+        TxnHash r_txn_hash;
+        bytes buff;
+
+        // Get the txnHash length and raw bytes of txnHash itself
+        size_t len;
+        // Loop through each txn in file
+        while (infile.read(reinterpret_cast<char*>(&len), sizeof(len))) {
+          infile.read(reinterpret_cast<char*>(&r_txn_hash), len);
+
+          // get the TxnReceipt length and raw bytes of TxnReceipt itself
+          infile.read(reinterpret_cast<char*>(&len), sizeof(len));
+          buff.clear();
+          buff.resize(len);
+          infile.read(reinterpret_cast<char*>(buff.data()), len);
+
+          // Deserialize the TxnReceipt bytes
+          TransactionWithReceipt r_tr;
+          r_tr.Deserialize(buff, 0);
+
+          if (r_tr.GetTransaction().GetTranID() != r_txn_hash) {
+            LOG_CHECK_FAIL("Txn Receipt Hash", r_txn_hash,
+                           r_tr.GetTransaction().GetTranID());
+            err = true;
+            continue;
+          }
+
+          BlockStorage::GetBlockStorage().PutTxBody(r_txn_hash, buff);
+          LOG_GENERAL(INFO, "Inserted Txn : " << r_txn_hash << endl);
+          if (saveToJsonFormat) {
+            Json::Value v = JSONConversion::convertTxtoJson(r_tr);
+
+            // create file with filename as "<<txnhash>>.json"
+            ofstream ofile(jsonOutputPath + "/" + r_txn_hash.hex() + ".json");
+            ofile << v.toStyledString();
+            ofile.close();
+          }
+        }
+      } catch (...) {
```

Comment:
instead of `...` can we just catch standard exception ?

Reply:
basically want to make sure we catch all kind of exception and program continues.
with `std::exception` can catch only those extending std::exception but no `runtime_error`  or any other. In that case needs to handle `runtime_exception` followed by `std::exception` and `...`.
so preferred this one over it.